src/util/simplify_expr_floatbv.cpp

```
@@ -20,9 +20,6 @@ Author: Daniel Kroening, kroening@kroening.com
 simplify_exprt::resultt<>
 simplify_exprt::simplify_isinf(const unary_exprt &expr)
 {
-  if(expr.op().type().id() != ID_floatbv)
```

I would leave that as a PRECONDITION.

---

Could do, but would be redundant with `to_floatbv_type(expr.type())` that `ieee_floatt::from_expr` does. I'm happy to add it for clarity, but if so I'd suggest to add it to `simplify_isnan` and `simplify_isnormal` as well?

---

Yes, please add, the idea being "fail early" and "clarity".

---

Done!